tokio-console/src/state/mod.rs

```diff
@@ -538,3 +538,35 @@ fn pb_duration(dur: prost_types::Duration) -> Duration {
     let nanos = u64::try_from(dur.nanos).expect("duration should not be negative!");
     Duration::from_secs(secs) + Duration::from_nanos(nanos)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_format_location() {
```

Comment:
This is a great idea. Could we also add tests for typical macOS (starting with `/Users/<username>`) and Windows (actually I don't know what this looks like) paths? Probably in separate test cases (separate `#[test]`).

Reply:
Good idea! I guess our current regex won't work for the Windows path. I'll test it on a Windows chip.

Reply:
<img width="1569" alt="image" src="https://github.com/tokio-rs/console/assets/29879298/79723d02-15b8-4efa-8e88-9481eefe82cc">
Actually, it works well on Windows. I will figure out how we handle it on Windows.

Reply:
I looked into the issue and found that the solution doesn't work for Windows paths. Although the output seems correct for this 'app' example, it actually isn't. The problem arises because when we use the `spawn` function, it uses `std::panic::Location::caller` to get the file location. I printed the location we received from the console-subscriber on Windows, which is `console-subscriber\\examples\\app.rs:68:10`. Unfortunately, this doesn't match the regex, so we end up calling `toString()`, which results in `console-subscriber\examples\app.rs:68:10`. 

To fix this problem, I will test more cases on Windows and improve the regex to support Windows path. But I don't think we should do it in this PR. I will open a new PR to fix it.
@hds What do you think? Does that make sense?

Reply:
Yep, that makes sense. Let's do it that way!